### DISCUSSION TITLE
Request: Is it possible to use Tauri instead of Electron?

### DISCUSSION THREAD

**Discussion by BDVGitHub** [author-more/penpot-desktop#48] [General]
Is it possible to use Tauri instead of Electron. In the install instructions, I saw that Electron is used. I know it will take a lot of work, but could you consider switching to Tauri? This is more performant in terms of security, results in less large install file, less memory consumption, a full comparison between Tauri and Electron can be found on the following site: [Tauri vs Electron](https://dev.to/dedsyn4ps3/goodbye-electron-hello-tauri-26d5)

**Comment by Belar** [comment]
Hello @BDVGitHub,
I’m open to use Tauri for Penpot Desktop and regularly check what’s the state of it. However, when Tauri brings compelling benefits, as you mentioned, the project so far benefited from their counterparts in Electron.

Electron is a mature ecosystem and a lot of problems have been solved by now. While it’s unfortunate that it ships a sizeable bundle, UI is uniform across platforms and feature availability is set. It has good embedding support (for now) and works well as a web app wrapper - something that is not priority of Tauri. Since I’m developing Penpot Desktop mostly myself, part-time, and sponsor its development, any time savings that can be achieved are precious from the perspective of the project.

I’m looking forward to giving Tauri another go. Current priorities allow us to not rush it though.

**Reply by BDVGitHub** [reply]
Understand I did some searching and bumped into the following GitHub repo via a post on reddit [Pake](https://github.com/tw93/Pake). This would allow converting a web page into a desktop app. Could this be a possible solution in relation to the web app wrapper missing from Tauri?

**Comment by bntmchr** [comment]
are there any new ideas, tests or what something else? is deno-app a valid solution for the penpot-app? actually is tauri a candidate for a desktop app-framework today? thank you for developing the penpot-desktop app ❤️

**Reply by Belar** [reply]
Hi @bntmchr,
You are welcome and thank you for the interest. 

Nothing is off the table, but at the moment I don’t have time to tackle a rewrite of the app. It made me consider expanding the use of LLMs to coding in this project, but I’m yet to decide.

Tauri was always the primary candidate. And the recently released Deno Desktop is interesting, definitely keeping an eye out on how it will develop. However, with each there will be trade-offs, hence why the original answer didn’t change much.
